server_fn/src/error.rs

```diff
@@ -31,24 +31,50 @@ impl From<ServerFnError> for Error {
     Clone,
     Copy,
 )]
-#[cfg_attr(
-    feature = "rkyv",
-    derive(rkyv::Archive, rkyv::Serialize, rkyv::Deserialize)
-)]
-pub struct NoCustomError;
+pub enum NoCustomError {}
+
+#[cfg(feature = "rkyv")]
+impl rkyv::Archive for NoCustomError {
+    const COPY_OPTIMIZATION: rkyv::traits::CopyOptimization<Self> =
+        rkyv::traits::CopyOptimization::disable();
+    type Archived = ();
+    type Resolver = ();
+    fn resolve(&self, _: Self::Resolver, _: rkyv::Place<Self::Archived>) {
+        match *self {}
+    }
+}
+
+#[cfg(feature = "rkyv")]
+impl<T, D: rkyv::rancor::Fallible + ?Sized> rkyv::Deserialize<T, D>
+    for NoCustomError
+{
+    fn deserialize(&self, _: &mut D) -> Result<T, D::Error> {
+        match *self {}
+    }
+}
+
+#[cfg(feature = "rkyv")]
+impl<S: rkyv::rancor::Fallible + ?Sized> rkyv::Serialize<S> for NoCustomError {
+    fn serialize(
+        &self,
+        _: &mut S,
+    ) -> Result<Self::Resolver, <S as rkyv::rancor::Fallible>::Error> {
+        match *self {}
+    }
+}
 
 // Implement `Display` for `NoCustomError`
 impl fmt::Display for NoCustomError {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "Unit Type Displayed")
+    fn fmt(&self, _: &mut fmt::Formatter) -> fmt::Result {
+        match *self {}
     }
 }
 
 impl FromStr for NoCustomError {
     type Err = ();
 
-    fn from_str(_s: &str) -> Result<Self, Self::Err> {
-        Ok(NoCustomError)
+    fn from_str(_: &str) -> Result<Self, Self::Err> {
+        Err(())
```

Comment:
Ironically, I do think `Infallible` would make sense here as the `Err` type for `FromStr` (since it implements `Error`), but this is a breaking change so not sure if its worth the effort.